internal/graph/check.go

```diff
@@ -1013,64 +1039,19 @@ func (c *LocalChecker) checkComputedUserset(ctx context.Context, req *ResolveChe
 	_, span := tracer.Start(ctx, "checkComputedUserset")
 	defer span.End()
 
-	seen := map[string]struct{}{}
-	typesys, _ := typesystem.TypesystemFromContext(ctx)
-
-	tk := tuple.NewTupleKey(
+	rewrittenTupleKey := tuple.NewTupleKey(
 		req.GetTupleKey().GetObject(),
-		req.GetTupleKey().GetRelation(),
+		rewrite.GetComputedUserset().GetRelation(),
 		req.GetTupleKey().GetUser(),
 	)
 
-	for {
-		tk = tuple.NewTupleKey(
-			tk.GetObject(),
-			rewrite.GetComputedUserset().GetRelation(),
-			tk.GetUser(),
-		)
-		key := tuple.TupleKeyToString(tk)
-		if _, cycleDetected := seen[key]; cycleDetected {
-			return func(ctx context.Context) (*ResolveCheckResponse, error) {
-				return &ResolveCheckResponse{
-					Allowed: false,
-					ResolutionMetadata: &ResolveCheckResponseMetadata{
-						CycleDetected: true,
-					},
-				}, nil
-			}
-		}
-		seen[key] = struct{}{}
-
-		userObject, userRelation := tuple.SplitObjectRelation(tk.GetUser())
-
-		// Check(document:1#viewer@document:1#viewer) will always return true
-		if tk.GetRelation() == userRelation && tk.GetObject() == userObject {
-			return func(ctx context.Context) (*ResolveCheckResponse, error) {
-				return &ResolveCheckResponse{
-					Allowed: true,
-					ResolutionMetadata: &ResolveCheckResponseMetadata{
-						DatastoreQueryCount: req.GetRequestMetadata().DatastoreQueryCount,
-					},
-				}, nil
-			}
-		}
-
-		rw, err := typesys.GetRelation(tuple.GetType(tk.GetObject()), tk.GetRelation())
-		if err != nil {
-			return func(ctx context.Context) (*ResolveCheckResponse, error) {
-				return nil, fmt.Errorf("relation '%s' undefined for object type '%s'", tk.GetRelation(), tuple.GetType(tk.GetObject()))
-			}
-		}
-		rewrite = rw.GetRewrite()
-		if _, isComputed := rewrite.GetUserset().(*openfgav1.Userset_ComputedUserset); !isComputed {
-			break
-		}
-	}
-
 	childRequest := clone(req)
-	childRequest.TupleKey = tk
+	childRequest.TupleKey = rewrittenTupleKey
 
-	return c.checkRewrite(ctx, childRequest, rewrite)
+	return func(ctx context.Context) (*ResolveCheckResponse, error) {
```

Comment:
🤔 Why are we reverting this?

Reply:
Because `checkRewrite` doesn't check for cycles, `ResolveCheck` does, and `ResolveCheck` eventually calls `checkRewrite`

Reply:
I like having the call to `hasCycle` in one place only :)